src/trainer.py

```diff
@@ -21,17 +26,28 @@ def train(self):
     self.model.to(self.device)
     wandb.watch(self.model)
     op = torch.optim.Adam(self.model.parameters(), lr=self.config.lr)
-    loss_f = torch.nn.BCEWithLogitsLoss()
+    if self.config.multi_class:
+      loss_f = torch.nn.CrossEntropyLoss()
+    else:
+      loss_f = diceloss()
+
+    metrics = {'pixel_acc': pixel_acc, 'per': precision, 'recall': recall,
+                   'dice': dice, 'iou': IoU}
 
     for epoch in range(self.config.n_epochs):
-      epoch_losses, mean_loss = self.train_epoch(op, loss_f)
-      dev_loss = self.evaluate(loss_f)
+      _, _ = self.train_epoch(op, loss_f)
+
+      dev_loss, dev_metrics = self.evaluate(loss_f, metrics)
```

Comment:
won't these break if `metrics` is none? Because in one case, evaluate returns a tuple, and otherwise it doesn't. Maybe it's safer to just always return a tuple, but if there are no metrics, it just returns an empty dict.